pkg/internal/appolly/appolly.go

```diff
@@ -34,26 +37,37 @@ type Instrumenter struct {
 
 	// tracesInput is used to communicate the found traces between the ProcessFinder and
 	// the ProcessTracer.
-	tracesInput *msg.Queue[[]request.Span]
+	tracesInput       *msg.Queue[[]request.Span]
+	processEventInput *msg.Queue[exec.ProcessEvent]
+	peK8SDecorator    swarm.InstanceFunc
 }
 
 // New Instrumenter, given a Config
 func New(ctx context.Context, ctxInfo *global.ContextInfo, config *beyla.Config) (*Instrumenter, error) {
 	setupFeatureContextInfo(ctx, ctxInfo, config)
 
 	tracesInput := msg.NewQueue[[]request.Span](msg.ChannelBufferLen(config.ChannelBufferLen))
+	processEventsInput := msg.NewQueue[exec.ProcessEvent](msg.ChannelBufferLen(10))
```

Comment:
I'm not sure if this is the best place to inject this additional small pipeline. @mariomac

Reply:
Yeah! I think so. I guess we can always move it to another part if later we require expanding the access.